packages/devp2p/src/eth/index.ts

```diff
@@ -208,7 +208,7 @@ export class ETH extends EventEmitter {
         this._latestBlock = status.latestBlock
       }
       const forkHashB = Buffer.from(this._forkHash.substr(2), 'hex')
-      const nextForkB = Buffer.from(this._nextForkBlock.toString(16), 'hex')
+      const nextForkB = int2buffer(this._nextForkBlock)
```

Comment:
I don't really see how this is a fix to a bug? What is changed here?

Reply:
I am just a simple man and still not so fluent in `Buffer` operations after all these years 😜 . The only thing I can say here is that the old code delivered an empty `Buffer`, just re-tested this with Node to be sure:

```shell
> let nextForkBlock = 5
undefined
> Buffer.from(nextForkBlock.toString(16), 'hex')
<Buffer >
```

Reply:
ha! here is some explanation: https://github.com/nodejs/node/issues/24491#issuecomment-440085063